### ISSUE TITLE
OpenSSH package in dom0 affected by CVE-2024-6387 (aka regreSSHion)

### ISSUE THREAD

**Issue by marmarek** [QubesOS/qubes-issues#9342]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release

R4.2

### Brief summary

OpenSSH server is not installed in dom0 by default. But when installed, it comes from Fedora 37 repositories at version 8.8p1 that is vulnerable to CVE-2024-6387.

Dom0 normally has no network access, so even if somebody does install OpenSSH server in dom0, it isn't exposed to the network. That can be changed by extra configuration (like using `qubes.ConnectTCP` service), but that is not recommended, and exposing dom0 network services to the internet is not security supported. There are several warnings about related actions, including:
- https://www.qubes-os.org/doc/how-to-install-software-in-dom0/
- warning about dangers of using `qubes.ConnectTCP` service in the [default policy](https://github.com/QubesOS/qubes-core-admin/blob/main/qubes-rpc-policy/90-default.policy):
   ```
   # WARNING: The qubes.ConnectTCP service is dangerous and allows any
   # qube to access any other qube TCP port. It should be restricted
   # only to restricted qubes. This is why the default policy is 'deny'
   ```  
and few more places

The only case where OpenSSH server in dom0, available from outside of dom0, was supported is qubes-remote-support package. That package was available in R4.1, but has never reached R4.2 stable repositories (although it is available in testing repositories) due to #8140. And even when installed (from testing repo), the tool is not functional due to the linked issue (when used, it will start the openssh server once, but fail at a later stage, making it not useful, and discouraging further use).

If `qubes-remote-support` tool would work, it configures the service to be available only from sys-whonix. And there, it can be reached only through an authenticated onion service, which provides additional protection layer. To attack OpenSSH in dom0, the attacker would need to either bypass authenticated onion service, or break into sys-whonix some other way. Nevertheless, if `qubes-remote-support` would work in R4.2, the OpenSSH issue would qualify for a Qubes Security Bulletin, as compromised sys-whonix should not allow breaking into dom0. 

Side note: currently presented attack works on a 32bit system. In its current form it cannot be performed on a 64bit system in a practical time frame. But it is believed it may be possible to develop an attack effective on a 64bit system too.

The OpenSSH version in R4.1 is not vulnerable to the issue in the title.

With all the above context, it's still useful to provide updated OpenSSH package for R4.2, even if it isn't supported configuration strictly speaking.

### Steps to reproduce

1. Install `openssh-server` package in dom0

### Expected behavior

Get a version not vulnerable.

### Actual behavior

Get a version 8.8p1 that is vulnerable to the issue in the title.

**Comment by alimirjamali** [comment]
Does this fix #7493 ?

**Comment by marmarek** [comment]
> Does this fix #7493 ?

Kinda yes
